androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/LauncherActivity.java

```diff
@@ -108,6 +109,42 @@
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
+        boolean hasNewTask = (getIntent().getFlags() & Intent.FLAG_ACTIVITY_NEW_TASK) != 0;
+        boolean hasNewDocument = (getIntent().getFlags() & Intent.FLAG_ACTIVITY_NEW_DOCUMENT) != 0;
+
+        if (!hasNewTask || hasNewDocument) {
+            // We only want one instance of the TWA running in the user's Recent Apps. This
+            // corresponds to ensuring that the TWA only exists in a single Task.
+
+            // If the TWA was implemented as single Activity, we could do this with
+            // launchMode=singleTask (or singleInstance), however since the TWA consists of a
+            // LauncherActivity which then starts a browser Activity, things get a bit more
+            // complicated.
+
+            // If we used singleTask on LauncherActivity then whenever a TWA was running and a new
+            // Intent was fired, the browser Activity on top would get clobbered.
+
+            // Therefore, we always ensure that LauncherActivity is launched with New Task. This
+            // means that if the TWA is already running a *new* LauncherActivity will be created on
+            // top of the Browser Activity. The browser then launches an Intent with CLEAR_TOP to
```

Comment:
It seems this code would force this new instance of LauncherActivity to restart as well.
Off the top of my head, I can't think of a way to distinguish LauncherActivity under TWA, mistakenly launched on top of something else, or a LauncherActivity on top of TWA that handles a new intent.

Reply:
Discussed offline.